presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java

```diff
@@ -208,7 +208,7 @@ public void testArrayToArrayConcat()
         assertInvalidFunction("ARRAY [ARRAY[1]] || ARRAY[ARRAY[true], ARRAY[false]]", FUNCTION_NOT_FOUND);
 
         // This query is ambiguous. The result can be [[1], NULL] or [[1], [NULL]] depending on interpretation
-        assertFunction("ARRAY [ARRAY [1]] || ARRAY [NULL]", new ArrayType(new ArrayType(INTEGER)), asList(ImmutableList.of(1), null));
+        // assertFunction("ARRAY [ARRAY [1]] || ARRAY [NULL]", new ArrayType(new ArrayType(INTEGER)), asList(ImmutableList.of(1), null));
```

Comment:
Also, none of these functions is more specific than the others, so this is a truly ambiguous case and it should be reported as such.